grakn-kb/src/main/java/ai/grakn/kb/internal/structure/AbstractElement.java

```diff
@@ -147,7 +147,13 @@ public boolean equals(Object object) {
     public void propertyUnique(P key, String value){
         if(!tx().isBatchTx()) {
             GraphTraversal<Vertex, Vertex> traversal = tx().getTinkerTraversal().V().has(key.name(), value);
-            if(traversal.hasNext()) throw PropertyNotUniqueException.cannotChangeProperty(element(), traversal.next(), key, value);
+
+            if(traversal.hasNext()){
+                Vertex vertex = traversal.next();
+                if(traversal.hasNext() || !vertex.equals(element())){
```

Comment:
We should flip this `||` for efficiency reasons (since the equals check is much faster)

Reply:
Good point